File: backend/app/models/leaderboard.py
"""리더보드 도메인 모델: LeaderboardEntry."""
from typing import Optional
from app.models.common import CamelModel


class LeaderboardEntry(CamelModel):
    id: str
    name: str
    email: str
    role: str
    avatar: Optional[str] = None
    today_score: int
    accuracy: float
    all_time_score: int

File: frontend/src/types/leaderboard.ts
/** 리더보드 도메인 타입: LeaderboardEntry 인터페이스. */
export type UserRole = "lead_analyst" | "acoustic_eng" | "data_analyst" | "junior_tagger" | "contractor";

export interface User {
  id: string;
  name: string;
  email: string;
  role: UserRole;
  avatar?: string;
  todayScore: number;
  accuracy: number;
  allTimeScore: number;
}
